File: backend/src/main/java/com/beni/backend/userAndFeedback/service/FeedbackService.java
package com.beni.backend.userAndFeedback.service;

import com.beni.backend.userAndFeedback.model.Feedback;
import com.beni.backend.userAndFeedback.model.Reply;
import com.beni.backend.userAndFeedback.repository.FeedbackRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service //
public class FeedbackService {

    @Autowired
    private FeedbackRepo feedbackRepo;         // Autowire the FeedbackRepo for interaction with the database


    // Method to add new feedback
    public Feedback addFeedback(Feedback feedback) {
        return feedbackRepo.save(feedback);
    }


    // Method to retrieve all feedbacks
    public List<Feedback> getAllFeedbacks() {
        return feedbackRepo.findAll();
    }


    // Method to retrieve feedback by its category
    public List<Feedback> getFeedbackByCategory(String category) {
        return feedbackRepo.findByCategory(category);
    }


    // Method to delete feedback by ID
    public boolean deleteFeedback(String id) {
        Optional<Feedback> feedback = feedbackRepo.findById(id);
        if (feedback.isPresent()) {
            feedbackRepo.deleteById(id);
            return true;
        }
        return false;
    }

    // Method to add a reply
    public Feedback addReplyToFeedback(String feedbackId, Reply reply) {
        Optional<Feedback> feedbackOpt = feedbackRepo.findById(feedbackId);
        if (feedbackOpt.isPresent()) {
            Feedback feedback = feedbackOpt.get();
            feedback.addReply(reply);
            return feedbackRepo.save(feedback);
        }
        return null; // if feedback not found
    }

}

File: frontend/src/app/feedback/services/feedbackService.ts
import axios from "axios";

const API_URL = "http://localhost:8080/api/feedback";

// Submit feedback
export const submitFeedback = async (feedback: any) => {
    return axios.post(API_URL, feedback);
};

// Get all feedback
export const getAllFeedback = async () => {
    return axios.get(API_URL);
};

// Get feedback by category
export const getFeedbackByCategory = async (category: string) => {
    return axios.get(`${API_URL}/category/${category}`);
};

// Delete feedback (Admin only)
export const deleteFeedback = async (id: string) => {
    return axios.delete(`${API_URL}/${id}`);
};

// Add a reply to feedback
export const addReply = async (feedbackId: string, reply: any) => {
    return axios.post(`${API_URL}/${feedbackId}/reply`, reply);
};
